### PR TITLE
Fix #275: Introduce parameters bag for container

### LINKED ISSUE #275: Introduce parameter bag for container configuration
Currently it's not convenient to configure container if we need definitions and one of the options. You have to specify other parameters.

```php
$config = ContainerConfig::create()->setDefinitions([...])->setStrict(true);
$container = new Container($config);
```

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -357,37 +364,36 @@ function (ContainerInterface $container): ContainerInterface
...
-    $tags,
-    $validate,
-    [
-        function (ContainerInterface $container): ContainerInterface {
-            // ...
-        }
-    ]
+    (new ContainerConfig)->withDelegates(
+        [
+            function (ContainerInterface $container): ContainerInterface {
```
Comment (vjik) [inline README.md:377]:
```suggestion
            static function (ContainerInterface $container): ContainerInterface {
```

File: src/Container.php [hunk 2]
```diff
@@ -245,7 +240,6 @@ private function setDefaultDefinitions(): void
      */
     private function setDelegates(array $delegates): void
     {
-        $this->delegates = new CompositeContainer();
```
Comment (vjik) [inline src/Container.php:248]: `setDelegates()` mean override current delegates. With this change `setDelegates()` always add delefates.

Maybe not delete `$this->delegates = new CompositeContainer();`, but in constructor use code:

```php
if ($config !== null) {
    $this->addProviders($config->getProviders());
    $this->setDelegates($config->getDelegates());
} else {
    $this->setDelegates([]);
}
```
        
or so:

```php
$config = $config ?? new ContainerConfig();
```

And set all parameters from config.

File: src/ContainerConfig.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * Container configuration.
+ */
+final class ContainerConfig implements ContainerConfigInterface
+{
+    private array $definitions = [];
+    private array $providers = [];
+    private array $tags = [];
+    private bool $validate = true;
+    private array $delegates = [];
+
```
Comment (rustamwin) [inline src/ContainerConfig.php:17]:
```suggestion

    private function __construct()
    {
    }
    
    public function create(): self
    {
        return new self();
    }
```

File: src/ContainerConfig.php [hunk 4]
```diff
@@ -15,6 +15,15 @@ final class ContainerConfig implements ContainerConfigInterface
     private bool $validate = true;
     private array $delegates = [];
 
+    private function __construct()
+    {
+    }
+
+    public static function create()
```
Comment (rustamwin) [inline src/ContainerConfig.php:22]:
```suggestion
    public static function create(): self
```

File: src/ContainerConfig.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    public function shouldValidate(): bool
+    {
+        return $this->validate;
+    }
+
+    /**
+     * @param array $delegates Container delegates. Each delegate is a callable in format
+     * "function (ContainerInterface $container): ContainerInterface". The container instance returned is used
```
Comment (vjik) [inline src/ContainerConfig.php:97]:
```suggestion
     * `function (ContainerInterface $container): ContainerInterface`. The container instance returned is used
```

File: src/ContainerConfigInterface.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    public function getTags(): array;
+
+    /**
+     * @return bool Whether definitions should be validated immediately.
+     */
+    public function shouldValidate(): bool;
+
+    /**
+     * @return array Container delegates. Each delegate is a callable in format
+     * "function (ContainerInterface $container): ContainerInterface". The container instance returned is used
```
Comment (vjik) [inline src/ContainerConfigInterface.php:34]:
```suggestion
     * `function (ContainerInterface $container): ContainerInterface`. The container instance returned is used
```

**Review by vjik — APPROVED** [review]

**Review by rustamwin — APPROVED** [review]

**Pull request merged**